server/src/test/java/org/elasticsearch/index/mapper/DateFieldMapperTests.java

```diff
@@ -152,7 +152,13 @@ protected List<ExampleMalformedValue> exampleMalformedValues() {
         return List.of(
             exampleMalformedValue("2016-03-99").mapping(mappingWithFormat("strict_date_optional_time||epoch_millis"))
                 .errorMatches("failed to parse date field [2016-03-99] with format [strict_date_optional_time||epoch_millis]"),
-            exampleMalformedValue("-522000000").mapping(mappingWithFormat("date_optional_time")).errorMatches("long overflow")
+            exampleMalformedValue("-522000000").mapping(mappingWithFormat("date_optional_time")).errorMatches("long overflow"),
```

Comment:
One not here is that due to the way this mapper is implemented trying to f.e. index an object actually hard fails and does not go via malformed code path. This feels like a bug since it works in other mappers. Fixing it is a breaking change though.

Reply:
Hm why is it a breaking change, we always return an error?

Reply:
When `ignore_malformed` is true we currently return an error but we'll stop if we fix the behavior.

Reply:
Isn't this the expectation? I'm trying to understand if it's a breaking change or a bug.

Reply:
I don't think this is a breaking change. If we ignore_malformed is set to true and we fail because of malformed data in whatever form then that was a bug.

Reply:
Sounds good, i'll create a bug for this.

Reply:
#109539